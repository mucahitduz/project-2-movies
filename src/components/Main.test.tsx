import { render, screen } from "@testing-library/react";
import Main from "./Main";
import "@testing-library/jest-dom";

describe("Main Component", () => {
  it("should render the children correctly", () => {
    render(
      <Main>
        <p>Test Child Content</p>
      </Main>
    );

    expect(screen.getByText("Test Child Content")).toBeInTheDocument();
  });

  it("should have the correct className", () => {
    render(
      <Main>
        <p>Test Child Content</p>
      </Main>
    );

    const mainElement = screen.getByRole("main");
    expect(mainElement).toHaveClass("main");
  });

  it("should match the snapshot", () => {
    const { asFragment } = render(
      <Main>
        <p>Test Child Content</p>
      </Main>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
