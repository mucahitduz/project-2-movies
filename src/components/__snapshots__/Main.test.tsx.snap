// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Component should match the snapshot 1`] = `
<DocumentFragment>
  <main
    class="main"
  >
    <p>
      Test Child Content
    </p>
  </main>
</DocumentFragment>
`;
